extensions/transferdetails/service.go

```diff
@@ -2,30 +2,32 @@ package transferdetails
 
 import (
 	"context"
+	"encoding/json"
 	"reflect"
 
 	"github.com/centrifuge/go-centrifuge/documents"
 	"github.com/centrifuge/go-centrifuge/errors"
 	"github.com/centrifuge/go-centrifuge/extensions"
 	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/jobs"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	logging "github.com/ipfs/go-log"
 )
 
 // Service defines specific functions for extension funding
 type Service interface {
 
-	// DeriveFromUpdatePayload derives TransferDetail from clientUpdatePayload
-	DeriveFromUpdatePayload(ctx context.Context, req UpdateTransferDetailRequest) (documents.Model, error)
+	// UpdateTransferDetail updates a TransferDetail
+	UpdateTransferDetail(ctx context.Context, req UpdateTransferDetailRequest) (documents.Model, jobs.JobID, error)
```

Comment:
I think in the go coding standard the package name should not be part of a method name.

We could rename it to `Update` and `Create`.